File: TP1/c++/Image.cpp
#include "Image.hpp"

Image::Image(int h = 10, int l = 10) {
	this->h = h;
	this->l = l;
	tab = new Pixel [this->h*this->l];
}

Image::~Image() {
	delete [] tab;
}

Pixel Image::get(int h, int l) {
	return tab[h*this->l + l];
}

void Image::set(int h, int l, int r, int g, int b) {
	this->get(h, l).set(r, g, b);
}

void Image::modifyEveryPixels(int nb_times) {
	for (int i=0 ; i<nb_times ; i++) {
		for (int j=0 ; j<this->h*this->l ; j++) {
			this->tab[j].set(j, 0, 0);
		}
	}
}

void Image::print() {
	for (int i=0 ; i<this->h*this->l ; i++) {
		if (i%this->l == 0)
			std::cout << std::endl;

		this->tab[i].print();
	}
}

File: TP1/java/Image.java
public class Image {
    int LARGEUR;
    int HAUTEUR;
    int TAILLE;
    Pixel[] image;

    public Image(int L, int H) {
        this.LARGEUR = L;
        this.HAUTEUR = H;
        this.TAILLE = L*H;
        this.image = new Pixel [this.TAILLE];
        
        for (int i=0 ; i<this.TAILLE ; i++) {
            this.image[i] = new Pixel();
        }
    }

    public Pixel get(int x, int y) {
        return this.image[x*this.LARGEUR + y];
    }

    public void set(int x, int y, int r, int g, int b) {
        this.get(x, y).set(r, g, b);
    }

    public void print() {
        for (int i=0 ; i<this.TAILLE ; i++) {
            if (i%this.LARGEUR == 0) {
                System.out.println();
            }
            System.out.print(this.image[i].getR() + " ");
        }
        System.out.println();
    }

    public void changeEveryPixels(int nb_times) {
        for (int j=0 ; j<nb_times ; j++)
            for (int i=0 ; i<this.TAILLE ; i++) {
                this.image[i].set(i, 0, 0);
            }
    }

}